src/style.css

```diff
@@ -0,0 +1,37 @@
+.body {
+  margin: 0;
+}
+
+.stars {
+  width: 100px;
+  display: flex;
+}
+
+.stars--star {
+  display: flex;
+  padding: 0;
+  width: 16px;
+  height: 16px;
+  background-image: url(images/star.svg);
+  background-repeat: no-repeat;
+  margin: 0 2px;
+  background-position: center;
+}
+
+.stars--1 .stars--star:nth-child(1),
+.stars--2 .stars--star:nth-child(1),
+.stars--2 .stars--star:nth-child(2),
+.stars--3 .stars--star:nth-child(1),
+.stars--3 .stars--star:nth-child(2),
+.stars--3 .stars--star:nth-child(3),
+.stars--4 .stars--star:nth-child(1),
+.stars--4 .stars--star:nth-child(2),
+.stars--4 .stars--star:nth-child(3),
+.stars--4 .stars--star:nth-child(4),
+.stars--5 .stars--star:nth-child(1),
+.stars--5 .stars--star:nth-child(2),
+.stars--5 .stars--star:nth-child(3),
+.stars--5 .stars--star:nth-child(4),
+.stars--5 .stars--star:nth-child(5) {
```

Comment:
there should be one selector for each modifier
for example all stars inside stars--5 should be yellow, we can achieve that with this selector `.stars--5 .stars__stars` (= selecting all stars__stars divs inside stars--5)
and for stars--3 only first 3 stars should be yellow, it's `.stars--3 .stars__item:nth-child(-n+3)` (= selecting first 3 stars__stars divs inside stars--3)

I suggest you read more about nth-child pseudoclass to understand the meaning of n argument inside it 🙂 
https://css-tricks.com/almanac/selectors/n/nth-child/
http://nthmaster.com/